Convert definitions to resources

ramereth opened the pull request:
This includes a variety of fixes most of which gets this ready for Chef 16 properly by converting the definitions to resources.

## Description

[Describe what this change achieves]

### Issues Resolved

[List any existing issues this PR resolves]

### Check List
- [ ] All tests pass. See https://github.com/sous-chefs/apache2/blob/master/TESTING.md
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

detjensrobert committed:
Initial conversion from apache recipes to modules
- change from including recipes to using resources
- change most attributes to their helper functions

detjensrobert committed:
fix template syntax error

detjensrobert committed:
force apache8+, test header match change

detjensrobert committed:
update nginx attributes to helper functions

detjensrobert committed:
add missing plugin dir

ramereth committed:
Cookstyle fixes

Ignoring the warnings with the definitions for now as that will be fixed later.

ramereth committed:
Cookstyle fixes

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Convert definitions to custom resources

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Switch to using apache2_mod_php resource

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Update nginx recipe to use newer nginx cookbook

Also remove nagios_test::fcgiwrap_rpm as it's not needed anymore on RHEL along
with the code which fixes the doc root permissions.

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Refactor and improve source installation

This fixes some issues with how the source installation works.

Namely, it removes the ability to add arbitrary patches as it created too many
issues with idempotency and really is outside the scope of this cookbook.

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Use apache_dir helper method instead of attribute

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Use proper locations for zapping distribution configuration

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Various fixes for pagerduty recipe

- Ensure we have server packages installed before running pager duty recipe
- Remove installation of perl Crypt-SSLeay package as it's no longer needed
- Add Perl Sys-Syslog package on RHEL as it's required (this is already
  installed on Debian-based systems from a dependency)

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Fix InSpec tests for swappable config suite

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Fix idempotency and other misc kitchen fixes

Also add testing for RHEL 8 and remove Ubuntu 16.04.

Signed-off-by: Lance Albertson <lance@osuosl.org>

kitchen-porter commented:
<!--
  0 Errors
  1 Warning: This is a big Pull Request.
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big Pull Request.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ramereth committed:
Switch to using stable chef to work around idempotency issues in the current channel

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Re-enable ChefSpec tests on packages

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Move nagios_conf template creation to delayed_action

This allows any of the nagios_* resources to be called after the inclusion of
the server recipe which is what actually generates the nagios config files.
Without this, any resources that were called after the recipe aren't added into
the template.

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Don't start nagios service as it will cause issues on initial runs on Ubuntu platforms

This works around a race condition on initial runs on Ubuntu platforms.

Signed-off-by: Lance Albertson <lance@osuosl.org>

detjensrobert reviewed (dismissed):
LGTM

tas50 commented on libraries/helpers.rb:
```diff
@@ -9,7 +9,7 @@ def nagios_vname
       elsif platform?('ubuntu')
         case node['platform_version'].to_f
-        when 16.04, 18.04
+        when 18.04
```
This whole thing could be improved to make this more future proof. Remove the case statement and just make it

```ruby
if node['platform_version'].to_f < 20.04
  'nagios3'
else 
  'nagios4'
end
```

Then it just magically works on 21.x

tas50 commented on test/smoke/swappable_config/default_test.rb:
```diff
@@ -1,4 +1,22 @@
+    case os.release.to_f
+    when 18.04
+      'nagios3'
+    when 20.04
```
This should just be > 20.04 here so these tests continue to work on newer versions

ramereth committed:
Fix logic so that Ubuntu >= 20.04 onward likely works

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth commented:
FYI I'm working on addressing an issue mentioned in d57c7a674b98d78f613ecb838f57a5515dd07f0d

kitchen-porter commented:
Released as: [10.0.0](https://github.com/sous-chefs/nagios/releases/tag/10.0.0)
